c7n/utils.py

```diff
@@ -47,30 +50,101 @@ class VarsSubstitutionError(Exception):
     pass
 
 
-def load_file(path, format=None, vars=None):
+class NoSuchS3Key(Exception):
+    pass
+
+
+class NoSuchS3Bucket(Exception):
+    pass
+
+
+ZIP_OR_GZIP_HEADER_DETECT = zlib.MAX_WBITS | 32
+
+
+class URIResolver:
```

Comment:
it really doesn't feel like resolver should be moving to utils, it was already in the resolver module

Reply:
importing it into `utils` created a circular dependency, only reason I moved it

Reply:
we don't really utils to become a dumping ground for bits, this was already cleanly split out in a separate module, ideally we resolve it in loader.py not in utils.py re avoiding circular dep.